cli/install_ee/install_ee.go

```diff
@@ -0,0 +1,170 @@
+package install_ee
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"regexp"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/tarantool/tt/cli/config"
+	"github.com/tarantool/tt/cli/util"
+	"github.com/tarantool/tt/cli/version"
+	"golang.org/x/term"
+)
+
+const eeSource string = "https://download.tarantool.io/enterprise/"
+
+type userCredentials struct {
+	username string
+	password string
+}
+
+// getCredsInteractive Interactively prompts the user for credentials.
+func getCredsInteractive() (userCredentials, error) {
+	res := userCredentials{}
+	reader := bufio.NewReader(os.Stdin)
+
+	fmt.Printf("Enter username: ")
+	resp, err := reader.ReadString('\n')
+	if err != nil {
+		return res, err
+	}
+	res.username = strings.TrimSpace(resp)
+
+	fmt.Printf("Enter password: ")
+	bytePass, err := term.ReadPassword(int(syscall.Stdin))
+	if err != nil {
+		return res, err
+	}
+	res.password = strings.TrimSpace(string(bytePass))
+
+	return res, nil
+}
+
+// getCredsFromFile gets credentials from file.
+func getCredsFromFile(path string) (userCredentials, error) {
+	res := userCredentials{}
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		return res, err
+	}
+
+	re := regexp.MustCompile("(?P<user>.*):(?P<pass>.*)")
+	matches := util.FindNamedMatches(re, strings.TrimSpace(string(data)))
+
+	if len(matches) == 0 {
+		return res, fmt.Errorf("Corrupted credentials.")
+	}
+
+	res.username = matches["user"]
+	res.password = matches["pass"]
+
+	return res, nil
+}
+
+// getCreds gets credentials for tarantool-ee download.
+func getCreds(cliOpts *config.CliOpts) (userCredentials, error) {
+	if cliOpts.EE == nil || (cliOpts.EE != nil && cliOpts.EE.CredPath == "") {
+		return getCredsInteractive()
+	}
+
+	return getCredsFromFile(cliOpts.EE.CredPath)
+}
+
+// getTarballName extracts tarball name from html data.
+func getTarballName(data string) (string, error) {
+	re := regexp.MustCompile(">(.*)<")
+
+	parsedData := re.FindStringSubmatch(data)
+	if len(parsedData) == 0 {
+		return "", fmt.Errorf("Cannot parse tarball name")
+	}
+
+	return parsedData[1], nil
+}
+
+// getVersions collects a list of all available tarantool-ee
+// versions for the host architecture.
+func getVersions(data *[]byte) ([]version.Version, error) {
```

Comment:
What is the format of input data? Variable name "data" does not give any clue and there is no description in comment. Is it HTML, or JSON?

Reply:
For now it is HTML, but it will be changed in nearest future, so it called `data` for now.